src/components/PlaybookProcess.js

```diff
@@ -144,29 +141,172 @@ class PlaybookProgress extends Component {
     });
   }
 
-  monitorSocket = (playId) => {
+  monitorSocket = (playbookName, playId, orderIndex) => {
     // Note that this function is only called after a fetch has completed, and thus
     // the application config has already completed loading, so getAppConfig can
     // be safely used here
     this.socket = io(getAppConfig('shimurl'));
     this.socket.on('playbook-start', this.playbookStarted);
-    this.socket.on('playbook-stop', this.playbookStopped);
-    this.socket.on('playbook-error', this.playbookError);
+    this.socket.on(
+      'playbook-stop',
+      (stepPlaybook) => { this.playbookStopped(stepPlaybook, playbookName, playId, orderIndex); });
+    this.socket.on(
+      'playbook-error',
+      (stepPlaybook) => { this.playbookError(stepPlaybook, playbookName, playId, orderIndex); });
     this.socket.on('log', this.logMessage);
-    this.socket.on('end', () => { this.socket.disconnect(); });
+    this.socket.on(
+      'end',
+      () => { this.processEndMonitorPlaybook(playbookName, playId, orderIndex); });
     this.socket.emit('join', playId);
   }
 
-  startPlaybook = () => {
-    // Start the playbook if it has not already been done.  overallStatus will be set
-    // initially here.  If the playbook is launched, further updates to the status will
-    // be performed as playbook events arrive from the ardana service (which originate
-    // in the ansible playbooks)
+  processEndMonitorPlaybook = (playbookName, playId, orderIndex) => {
```

Comment:
remove orderIndex from this processing and all processing in this file.

Reply:
k